scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -406,7 +406,7 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
   def hashPartition(numPartitions: Int): Seq[SCollection[T]] =
     self.partition(
       numPartitions,
-      t => Math.floorMod(t.hashCode(), numPartitions)
+      t => Math.floorMod(ScioUtil.hashCodeFn[K](t), numPartitions)
```

Comment:
should be `T` right?

Reply:
yes absolutely, I am now convinced that my coding skills are bad without a compiler!!!